lib/checkunusedvar.cpp

```diff
@@ -1609,17 +1609,31 @@ bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const To
     }
 
     bool sideEffectReturnFound = false;
+    std::list<const Variable*> pointersToGlobals;
     for (Token* bodyToken = func.functionScope->bodyStart->next(); bodyToken != func.functionScope->bodyEnd;
-         bodyToken = bodyToken->next()) {
+            bodyToken = bodyToken->next())
+    {
+        // check variable inside function body
         const Variable* bodyVariable = bodyToken->variable();
         if (bodyVariable) {
-            // check variable for side-effects
             if (!isVariableWithoutSideEffects(*bodyVariable)) {
                 return false;
             }
             // check if global variable is changed
-            if (bodyVariable->isGlobal()) {
-                return false; // TODO: analyze global variable usage
+            if (bodyVariable->isGlobal() ||
+                (std::find(pointersToGlobals.begin(), pointersToGlobals.end(), bodyVariable) != pointersToGlobals.end()))
+            {
+                if (isVariableChanged(bodyToken, 20, nullptr, true)) {
```

Comment:
indirect=20.. hmm is that a test?

I believe you should pass `mSettings` as third argument. And `mTokenizer->isCPP()` as 4th argument.

Reply:
> indirect=20.. hmm is that a test?

I don't know what value should be used here. I found that search depth default value in `astutils.h` is 20, so just used it.
Should I use another value?

Reply:
ok it should be fine.. thanks.